packages/blocks/shuffle/src/reducer.ts

```diff
@@ -0,0 +1,93 @@
+import { Reducer } from "react";
+import produce from "immer";
+import { v4 as uuid } from "uuid";
+
+export type Item = {
+  id: string;
+  value: string;
+};
+
+export type Items = Item[];
+
+type Action<S> = {
+  type: S;
+};
+
+type ActionWithPayload<S, T> = Action<S> & {
+  payload: T;
+};
```

Comment:
You can give the generic a default of `void` to allow it to be undefined, and then add to the type if it has been defined. Then you can make all the actions have type `Action`
```suggestion
type Action<S, T = void> = {
  type: S;
} & (T extends void ? {} : { payload: T });
```

Reply:
Thanks! Addressed in 3bf8759a0afc9eb9f265038677d0ca19d5d5d66f